app/models/form.rb

```diff
@@ -175,9 +175,10 @@ def visible_questionings
     questionings.reject{|q| q.hidden?}
   end
 
-  # returns questionings that work with sms forms and are not hidden
+  # returns hash of questionings that work with sms forms and are not hidden
   def smsable_questionings
-    questionings.reject{|q| q.hidden? || !q.question.qtype.smsable?}
+    smsable_questionings = questionings.reject{|q| q.hidden? || !q.question.qtype.smsable?}
+    smsable_questionings.index_by.with_index { |_, i| i + 1 }
```

Comment:
Neither do I :smile_cat: I'm just taking for granted that, however enumerators work, they manage to call the block correctly even when passed through a chain.